### ISSUE TITLE
Import does not publish the content

### ISSUE THREAD

**Issue by akkina9** [contentful/contentful-cli#127]
I am trying to import content (some of it should be published and some should be in draft mode). When I import the data into contentful, they are by default in draft mode. Should I be sending some additional info in the JSON for me to force publish? Should I be using some of the publish* fields?

Here is the sample input data:
{
  "entries": [
    {
      "sys": {
        "space": {
          "sys": {
            "type": "Link",
            "linkType": "Space",
            "id": "fadfjadlkfad"
          }
        },
        "type": "Entry",
        "contentType": {
          "sys": {
            "type": "Link",
            "linkType": "ContentType",
            "id": "category"
          }
        }
      },
      "fields": {
        "title": {
          "en-US": "Hello World Category"
        },
        "slug": {
          "en-US": "hello-world-category"
        }
      }
    }
  ]
}

**Comment by scaccogatto** [comment]
I am sorry, does anyone have a solution for this?

Example:

```
{
    "entries": [
        {
            "sys": {
                "publishedVersion": 1,
                "space": {
                    "sys": {
                        "type": "Link",
                        "linkType": "Space",
                        "id": "4zg16q5vss6b"
                    }
                },
                "environment": {
                    "sys": {
                        "id": "staging",
                        "type": "Link",
                        "linkType": "Environment"
                    }
                },
                "contentType": {
                    "sys": {
                        "type": "Link",
                        "linkType": "ContentType",
                        "id": "model"
                    }
                }
            },
            "fields": {
                "tesyId": {
                    "it-IT": "0049130"
                },
                "synced": {
                    "it-IT": false
                }
            }
        }
    ]
}
```

**Comment by phoebeschmidt** [comment]
Hi @scaccogatto 

Is there already content in the space you're importing into?

You can read more about the publishing strategy of the import tool when there is existing content here: https://github.com/contentful/contentful-import#bulb-importing-to-a-space-with-existing-content

The `sys` properties are ignored, so sending published data in the `sys` payload will not force a publish.

**Comment by scaccogatto** [comment]
I solved by using the cli manually, so I don't need this feature anymore.
If it helps, yes, there is already content in the space.

**Comment by phoebeschmidt** [comment]
glad you were able to solve it. 

The existing content in the space, and its status (published, draft, etc) was likely having an effect on whether the import tool published the content.

**Comment by scaccogatto** [comment]
I am sorry, this was 10 day ago. I forgot almost everything. I did NOT solve this problem with the cli.
I cannot make it publish after importing.
The content imports correctly but it tets itself to draft.
There is more content in the space but not the one that I am importing. I am stuck.

**Comment by phoebeschmidt** [comment]
@scaccogatto could you open a new issue and we can continue debugging there?

**Comment by nmercado-modus** [comment]
Hi, @scaccogatto did you get any feedback on this? I'm facing the same issue. Imported assets are set as drafts.

Thanks in advance.

**Comment by scaccogatto** [comment]
> Hi, @scaccogatto did you get any feedback on this? I'm facing the same issue. Imported assets are set as drafts.
> 
> Thanks in advance.

I had to rewrite my logic with the contentful js sdk.
Never had the chance to write the whole thing as I wanted because of this bug.

**Comment by KaterynaBoiko** [comment]
There's the following piece of code in the library. It filters all the entities and decides if they should be published or not:
```
const entityIdsToPublish = sourceEntities
    .filter(({ original }) => original.sys.publishedVersion)
    .map(({ original }) => original.sys.id)
```
Only after finding this, it get clear that if an entity has to be published, it should contain ```sys.publishedVersion``` field

**Comment by dsacramone** [comment]
> There's the following piece of code in the library. It filters all the entities and decides if they should be published or not:
> 
> ```
> const entityIdsToPublish = sourceEntities
>     .filter(({ original }) => original.sys.publishedVersion)
>     .map(({ original }) => original.sys.id)
> ```
> 
> Only after finding this, it get clear that if an entity has to be published, it should contain `sys.publishedVersion` field

@KaterynaBoiko 

Is there anything else to add? I did add the "publishedVersion" and it didn't work.  Does this look correct?

This is a test json, and it works. It gets put in a "Draft" status. It does NOT publish. 

```
{
  "entries": [
{
  "sys": {
      "contentType": {
        "sys": {
          "type": "Link",
          "linkType": "ContentType",
          "id": "localCity"
        }
      },
      "publishedVersion": 1
    },
    "fields": {
      "name": {
        "en-US": "Venice, California"
      },
      "seoName": {
        "en-US": "venice"
      },
      "state": {
        "en-US": {
          "sys": {
            "type": "Link",
            "linkType": "Entry",
            "id": "bfNl4hfwlTsOxgg6NfAqh"
          }
        }
      },
      "stateSlug": {
        "en-US": "ca"
      },
      "richTest": {
        "en-US": {
          "nodeType": "document",
          "data": {
          },
          "content": [
            {
              "nodeType": "paragraph",
              "content": [
                {
                  "nodeType": "text",
                  "value": "THIS IS A VENICE TEXT",
                  "marks": [
                    {
                      "type": "bold"
                    }
                  ],
                  "data": {
                  }
                }
              ],
              "data": {
              }
            },
            {
              "nodeType": "paragraph",
              "content": [
                {
                  "nodeType": "text",
                  "value": "hello bro VENICE YOYOYOYOYOY ",
                  "marks": [
                  ],
                  "data": {
                  }
                }
              ],
              "data": {
              }
            },
            {
              "nodeType": "paragraph",
              "content": [
                {
                  "nodeType": "text",
                  "value": "<p>testing paragraph tag in Venice</p>",
                  "marks": [
                  ],
                  "data": {
                  }
                }
              ],
              "data": {
              }
            }
          ]
        }
      },
      "textTest": {
        "en-US": "<p>test</p>\n<a href='http://www.venice.com/'>click me</a> "
      }
    }
}]
}
```
